substrate/frame/referenda/src/migration.rs

```diff
@@ -93,6 +95,21 @@ pub mod v1 {
 	/// The log target.
 	const TARGET: &'static str = "runtime::referenda::migration::v1";
 
+	pub(crate) type ReferendumInfoOf<T, I> = ReferendumInfo<
+		TrackIdOf<T, I>,
+		PalletsOriginOf<T>,
+		SystemBlockNumberFor<T>,
+		BoundedCallOf<T, I>,
+		BalanceOf<T, I>,
+		TallyOf<T, I>,
+		<T as frame_system::Config>::AccountId,
+		ScheduleAddressOf<T, I>,
+	>;
+
+	#[storage_alias]
+	pub type ReferendumInfoFor<T: Config<I>, I: 'static> =
+		StorageMap<Pallet<T, I>, Blake2_128Concat, ReferendumIndex, ReferendumInfoOf<T, I>>;
```

Comment:
Personally I would move it to the other module. And consider the V1 storage to be with the provided block number.

Reply:
But why? isn't the reason why this was declared here is to tell that this Type and Storage was supported in v1?

Reply:
let's consider we implement this as storage version 2.

I have my pallet in my chain in storage version 1.
I update the version of my pallet without changing the block number provider, the migration executes don't change the storage and write storage version 2.
Then I want to change the `BlockNumberProvider`.
I can't reuse the logic, the pallet is already in version 2.

Reply:
but maybe it is better to keep this storage here.

If the type `BlockNumber` is changing, they would want to do the migration `v0 -> v1 -> migrate_block_number_provider`
If we change the type here to the provided block number then they will have to write a manual migration.

Maybe you were right when implementing with v2. And we should have both a v1 to v2 migration and a `migrate_block_number_provider` standalone migration.
But because everybody use `u32` for block number I think having `v0 -> v1 -> migrate_block_number_provider` is ok.